### ISSUE TITLE
Support for non-diagonal k-point meshes

### ISSUE THREAD

**Issue by brainandforce** [xamberl/DFT-raMO#42]
Currently, a lot of the code assumes that the k-point mesh is represented by a diagonal matrix, but in reality, the k-point mesh is actually a matrix which may have off-diagonal components. It can be advantageous to use those in some contexts (for example, unit cells that are not orthorhombic), so I'm opening this issue to see if we can treat k-point meshes more rigorously.

**Comment by xamberl** [comment]
DFT-raMO requires a large supercell to fold the k-points of the Brillouin zone onto the gamma point. The k-point mesh for the preparatory calculations must correspond to this large supercell.

It is mathematically possible to generate a large orthorhombic supercell from a non-orthogonal unit cell. However, I think we also will need to consider how a user will figure out what k-point mesh to use. I don't think this is a priority at the moment.

**Comment by brainandforce** [comment]
> I don't think this is a priority at the moment.

Agreed, I figure it's good to open this issue, but I've labeled it as an enhancement as not critical at the moment.
